cmake/Modules/FindMKL.cmake

```diff
@@ -43,55 +43,7 @@ endif()
 # ---[ Root folders
 set(INTEL_ROOT "/opt/intel" CACHE PATH "Folder contains intel libs")
 
-if(USE_MKLDNN)
```

Comment:
Though now it is convoluted in 1 file there is a configuration difference between full MKL and MKLML. For example, the `librt` library is present only in full MKL, OpenMP library needs to be configured explicitly for MKLML. Previously, I made an attempt to separate it to 2 different files, maybe you can reuse some code from there [#11148](https://github.com/apache/incubator-mxnet/pull/11148/files)

Reply:
@Chancebair . @lebeg  thank you a lot for the review.   Yes, we actually need full MKL package install on CI machine.  There are mainly two factors , which 
1)  MKLDNN will don't depend on the MKLML in the future,
2) second, but important, MXNET will use more MKL functions other than in MKLML. 
So we decide to unlock MKLDNN (MKLML) and MKL dependency, which means, 
use MKLDNN and USE_BLAS=mkl completely independent. 
1)MKLDNN is on, it will download MKLML automatically. doesn't use full MKL package. 
2)  when USE_BLAS=mkl,  only use the full MKL. (MKL_ROOT). won't detect  MKLML (from MKLDNN). 

Thanks